node/src/node.rs

```diff
@@ -66,14 +77,133 @@ where
             p2p: p2p.clone(),
         })?);
 
-        Ok(Node { p2p, syncer })
+        Ok(Node { p2p, store, syncer })
     }
 
+    /// Grab a reference to the [`P2p`] component of the node.
+    #[doc(hidden)]
     pub fn p2p(&self) -> &P2p<S> {
         &self.p2p
     }
 
+    /// Grab a reference to the [`Syncer`] component of the node.
+    #[doc(hidden)]
     pub fn syncer(&self) -> &Syncer<S> {
         &self.syncer
     }
+
+    /// Grab a reference to the [`Store`] component of the node.
+    #[doc(hidden)]
+    pub fn store(&self) -> &S {
+        &self.store
+    }
```

Comment:
I'm not sure on why we should keep this

Reply:
I was wondering whether to remove it. There are some left end to end tests that make the raw header-ex requests but we can just delete them. Up to you

Reply:
removed